src/workerd/api/tests/js-rpc-test.js

```diff
@@ -1031,6 +1063,48 @@ export let streams = {
   }
 }
 
+export let serializeHttpTypes = {
+  async test(controller, env, ctx) {
+    {
+      let headers = await env.MyService.returnHeaders();
+      assert.strictEqual(headers instanceof Headers, true);
+
+      // Awkwardly, there's actually no API to get the non-lowercased header names.
+      assert.strictEqual([...headers].map(pair => pair.join(": ")).join("\n"),
+          "content-length: 123\n" +
+          "corge: !@#\n" +
+          "foo: bar\n" +
+          "set-cookie: abc\n" +
+          "set-cookie: def");
```

Comment:
does `assert.deepStrictEqual(...)` not work here? e.g. 

```js
const check = new Headers([[ ... expected headers ]]);
assert.deepStrictEqual(check, headers);
```

I know it works in Node.js to compare map-like objects like `Headers` but I can't remember if it works with our polyfills.

Reply:
Unfortunately, `assert.deepStrictEqual(a, b)` appears to pass if the parameters are merely both of type `Headers`, regardless of whether their content matches.

That said, using deep-equal to compare `[...headers]` against an array does work and is a lot nicer than my weird joins here. 5615cde4